### PR TITLE
Allow setting band-specific scale and offset values

### PR DESCRIPTION
**Pull request by bleutner**
Previously, only a single scale and/or offset value could be set by gdal_edit.py, which was applied to all layers. Now, one can also specify one value per band, like this:

`gdal_edit.py -scale 1 10 100 -offset 0 -10 10 ThreeLayerFile.tif`

This functionality is useful for files with different scaling factors, which can, for example, be produced with `gdal_translate -scale_1 -scale_2 ...`

### CODE REVIEW

File: gdal/swig/python/scripts/gdal_edit.dox [hunk 1]
```diff
@@ -87,7 +87,9 @@ Assign a specified nodata value to output bands.</dd>
 Remove existing nodata values.</dd>
 
 <dt> <b>-scale</b> <i>value</i>:</dt><dd> (GDAL >= 2.2)
-Assign a specified scale value to output bands. If no scale is needed, it
+Assign a specified scale value to output bands. If a single scale value is provided it will be 
+set for all layers. Alternatively one scale value per band can be provided, in which case the
```
Comment (rouault) [inline gdal/swig/python/scripts/gdal_edit.dox]: layers -> bands

File: gdal/swig/python/scripts/gdal_edit.dox [hunk 2]
```diff
@@ -96,7 +98,9 @@ true_value = (pixel_value * scale) + offset
 Note: these values can be applied using -unscale during a gdal_translate run.</dd> 
 
 <dt> <b>-offset</b> <i>value</i>:</dt><dd> (GDAL >= 2.2)
-Assign a specified offset value to output bands. If no offset is needed, it
+Assign a specified offset value to output bands. If a single offset value is provided it will be 
+set for all layers. Alternatively one offset value per band can be provided, in which case the
```
Comment (rouault) [inline gdal/swig/python/scripts/gdal_edit.dox]: layers -> bands

File: gdal/swig/python/scripts/gdal_edit.py [hunk 3]
```diff
@@ -261,6 +268,22 @@ def gdal_edit(argv):
     if ds is None:
         return -1
 
+    if scale:
+        if len(scale) == 1:
+            scale = scale * ds.RasterCount 
+        elif len(scale) != ds.RasterCount:
+            print('If more than one scale value is provided, their number must match the number of layers.')
```
Comment (rouault) [inline gdal/swig/python/scripts/gdal_edit.py]: layers -> bands

File: gdal/swig/python/scripts/gdal_edit.py [hunk 4]
```diff
@@ -261,6 +268,22 @@ def gdal_edit(argv):
...
+            print('')
+            return Usage()
+    
+    if offset:
+        if len(offset) == 1:
+            offset = offset * ds.RasterCount
+        elif len(offset) != ds.RasterCount:
+            print('If more than one offset value is provided, their number must match the number of layers.')
```
Comment (rouault) [inline gdal/swig/python/scripts/gdal_edit.py]: layers -> bands

File: gdal/swig/python/scripts/gdal_edit.dox (hunk 1 above)
Comment (bleutner) [reply gdal/swig/python/scripts/gdal_edit.dox]: changed "layers" to "bands"

**Pull request merged**